src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/visualizers/RequestViewHTTP.java

```diff
@@ -316,9 +324,11 @@ public static Map<String, String[]> getQueryMap(String query) {
                 System.arraycopy(known, 0, tmp, 0, known.length);
                 known = tmp;
             }
-            map.put(name, known);
+            if (!noNameAndNoValue) {
```

Comment:
Double negations are hard to read.
Perhaps we could use `validNameAndValue` with a default of `true` and set it to `false`, if `name` or `value` are missing.